eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -284,11 +284,13 @@ object LightningMessageCodecs {
     ("chainHash" | binarydata(32)) ::
       ("shortChannelId" | shortchannelid) ::
       ("timestamp" | uint32) ::
-      ("flags" | binarydata(2)) ::
+      ("messageFlags" | binarydata(1)) ::
+      ("channelFlags" | binarydata(1)) ::
       ("cltvExpiryDelta" | uint16) ::
       ("htlcMinimumMsat" | uint64) ::
       ("feeBaseMsat" | uint32) ::
-      ("feeProportionalMillionths" | uint32))
+      ("feeProportionalMillionths" | uint32) ::
+      ("htlcMaximumMsat" | optional(bool, uint64)))
```

Comment:
That's not compliant: this encoding will prefix the `htlc_maximum_msat` field by a bit telling if there is data or not. Instead the bit should be set in the `message_flags` field.

One way to do it is to add a `require` in the `ChannelUpdate` constructor that makes sure that the flag is set when the optional field is defined, and use a `conditional` codec with a `>>:~` operator, à la:

https://github.com/scodec/scodec-protocols/blob/6b33ffcea6da43d499ee455f201db9389476d51e/src/main/scala/scodec/protocols/ip/tcp/TcpHeader.scala#L37-L49

Let me know if you need help with this.

Reply:
Btw this bit is also what causes the test error in `FailureMessageCodecsSpec`.

Reply:
Thanks, I've decided to use `flatAppend` because `>>:~` produces an `hlist` of different structure.